### PR TITLE
Geant4 BFC options

### PR DESCRIPTION
**Pull request by klendathu2k**
Adding BFC options to load geant4 libraries, and load the relevant spack modules to support.   (The latter is better added to the star-env modules?)

### CODE REVIEW

**Comment by klendathu2k** [comment]
Not quite ready for prime time... seems to break several tests.  Investigating.

**Comment by klendathu2k** [comment]
In order to test functionality, need to load G4 modules via
```
$ module load geant4-10.4.0                  
$ module load geant4-data-10.4.0 # may not be required
```

File: StRoot/StBFChain/BigFullChain.h [hunk 1]
```diff
@@ -1410,7 +1422,7 @@ Bfc_st BFC[] = { // standard chains
    ,                                "Pythia6_4_26,bpythia","Load Pythia in starsim, set pp 510 GeV",kFALSE},
   {"Wenu"        ,"" ,"","pythia","", ""                 ,"set pp 510 GeV -> W+/- -> e+/- nu/nubar",kFALSE},
   {"hijing"      ,"geant" ,"","-gstar,-fzin,-ntin,-geant,geantL","St_geant_Maker"
-   ,"StarGeneratorUtil,StarGeneratorEvent,StarGeneratorBase,Hijing1_383"
+   ,"StarGeneratorUtil,StarGeneratorEvent,StarGeneratorBase,Hijing1_383"
```
Comment (plexoos) [inline StRoot/StBFChain/BigFullChain.h]:
```suggestion
   ,"StarGeneratorUtil,StarGeneratorEvent,StarGeneratorBase,Hijing1_383"
```

**Review by plexoos — APPROVED** [review]

**Comment by klendathu2k** [comment]
Any thoughts as to why the whitespace in BigFullChain.h was causing runtime errors?

**Comment by plexoos** [comment]
The white space change was used to trigger CI. There was a small adjustment in the container environment

**Review by genevb — APPROVED** [review]
No objections.

**Comment by klendathu2k** [comment]
And... we merge.

**Pull request merged**